### ISSUE TITLE
Fix all the files that fail to compare correctly

### ISSUE THREAD

**Issue by est31** [RustAudio/lewton#26]
Commits 3224e949ff2fbed3e17e6abd31ad6558a4867a4f...6352249e535fdb1e8cf84b54829fa4c17824f14d have added files to the testsuite from `libnogg` as well as from `xiphmont`. Some are commented out however, as lewton shows mismatches to libvorbis.

xiph testsuite failures (27 test files in total):

- [x] `chain-test1.ogg` (dbfaae5ad07edf322a7e41a43b2ecc2f7504f8ed)
- [x] `chain-test2.ogg` (e051c7efba1e023e7cdcf6e64396fb7a8ff66947)
- [x] `chain-test3.ogg`(e2b38e849bd51e21a7c30baff7d5f2d0bd9d0653)
- [x] `singlemap-test.ogg` (1073e58b1c36bf77993dbd08433c4a4b61acd760)
- [x] `test-short.ogg` (e2b38e849bd51e21a7c30baff7d5f2d0bd9d0653)
- [x] `unused-mode-test.ogg` (0517494f9a4849c77c474b988471d2085fca7a81)

libnogg testsuite failures (32 test files in total):

- [x] `6ch-all-page-types.ogg` (6b0a2f8d18c152cef0a02d143d23e470ef6db0ab, 47181325b27456d3862e4a44e0979d3f4c4aea06)
- [x] `6ch-long-first-packet.ogg` (6b0a2f8d18c152cef0a02d143d23e470ef6db0ab, 47181325b27456d3862e4a44e0979d3f4c4aea06)
- [x] `6ch-moving-sine-floor0.ogg` (71190ca6d0c4e92f5b25ec2615016b1f5951efd9)
- [x] `6ch-moving-sine.ogg`(6b0a2f8d18c152cef0a02d143d23e470ef6db0ab)
- [x] `bad-continued-packet-flag.ogg` (95fa236fb87e8f00e531bee710f3c3522530e639)
- [x] `noise-6ch.ogg` (6b0a2f8d18c152cef0a02d143d23e470ef6db0ab)
- [x] `single-code-2bits.ogg` (95fa236fb87e8f00e531bee710f3c3522530e639)
- [x] `single-code-nonsparse.ogg` (10c479d9bafa5181ae1f863a3361dd6abf5d2439)
- [x] `single-code-ordered.ogg` (b312ce46ebee50505ef529223826f99ef81d9be6)
- [x] `single-code-sparse.ogg` (can decode with b312ce46ebee50505ef529223826f99ef81d9be6, correct samples with 6b0a2f8d18c152cef0a02d143d23e470ef6db0ab)
- [ ] `square-interleaved.ogg`
- [x] ~`square-with-junk.ogg`~ (eb231f72d90e01fa2c0bab59e6cda87a38bf4078) → actually an ogg crate issue https://github.com/RustAudio/ogg/issues/7

**Comment by est31** [comment]
So, as of the 0.9.1 release, only these 5 files are remaining. I ran `stb_vorbis` on them. For some, it sadly considered the files as invalid and returned `VORBIS_invalid_setup`.

* chain-test1.ogg → here, the "first" stream in the chain has no error, and I think `stb_vorbis` supports no chaining. So for further investigation, I'd have to split the file up.
* chain-test3.ogg →  `VORBIS_invalid_setup`
* singlemap-test.ogg → mismatch in residue
* test-short.ogg → `VORBIS_invalid_setup`
* square-interleaved.ogg →  `VORBIS_invalid_setup`

**Comment by est31** [comment]
With 1073e58b1c36bf77993dbd08433c4a4b61acd760, singlemap-test.ogg has been fixed. 
As of the 0.9.2 release, only 4 files are thus remaining.
I ran further tests on them.

* chain-test1.ogg → I've split up the file via opening it with hexdump -C | less and then searching for stream starts by /-ing for  `vorbis`, looking for the `OggS` preceeding it, then taking this as an offset to discard the bytes via dd. Basically `dd if="chain-test1.ogg" of="chain-test-1-first-part.ogg" bs=1 count=565344` and `dd if="chain-test1.ogg" of="chain-test-1-second-part.ogg" bs=1 skip=565344`. The results were interesting! Apparently there is no mismatch either of the two parts alone, only in the combined version. We have a mono file followed by a stereo file, maybe this is the reason? No idea.

* chain-test3.ogg → I've split up this file as well (at offs `29781` this time). Here we have a stereo file followed by a mono one. The comparison of the combined stream already fails at packet 1, while the first part in isolation fails to compare, and the second part in isolation compares successfully (without a difference). If I play back the first file in a player it sounds just like `test-short.ogg` so I suspect it's the same bug.

* test-short.ogg → libvorbis outputs samples for the first packet, while lewton outputs only silence. Residue post inverse values look reasonable (non-zero values), but the post dot product pre imdct values have tons of NaNs, inf, 0, and values close to 0. test-short.ogg seems to be using floor type zero.

* square-interleaved.ogg → Opening the file with the `format-info` example of libvorbis indicates that the file is indeed interleaved. The ogg spec has some words on interleaved streams, but I still am not 100% sure what to do here. Maybe testing a bit will help.
